### ISSUE TITLE
Deleting a user:// folder with an invalid path deletes top level project files

### ISSUE THREAD

**Issue by VizuaaLOG** [godotengine/godot#40146]
**Godot version:**
3.2.2

**OS/device including version:**
Windows 10 64bit

**Issue description:**
When using the Directory class to delete folders/files within the user:// path if a variable is used to concatenate paths together and the variable has not been defined, rather than failing Godot seems to delete the top level files within the project folder. I believe including a typo without a var does not appear to produce this issue.

I came across this issue while working on save deletions within my game, I believe the variable I'm using is defined, however this is the only way I've been able to reproduce this in a minimum project.

My assumption is the user:// path within Godot doesn't know about or have access to the project files and so this is very odd behaviour.

**Steps to reproduce:**
1. Create new files/folders within the user:// directory
2. Delete the files/folders and include an undefined variable to concatenate the paths
3. Watch the project files get deleted :(

**Minimal reproduction project:**
[Delete-Test.zip](https://github.com/godotengine/godot/files/4875846/Delete-Test.zip)

I've managed to reproduce this minimal project, it includes three buttons:
- The first will setup a basic folder structure with some dummy files inside
- The second will try and delete those files using an undefined `name` variable, clicking this button will delete some of the files and also the project's own source files
- The third button will do the same as the second but with the `name` variable defined, this would have the expected behaviour

**Comment by nathanfranke** [comment]
The `name` variable isn't undefined. You're accessing the `name` property of `Node` (The node's name)

There is still a bug here, but it's a lot simpler. Try this:

```py
var dir = Directory.new()
dir.open("a") # Invalid path

dir.list_dir_begin(true)

while true:
	var file = dir.get_next()
	if file == "":
		break
	
	print(file)

dir.list_dir_end()
```

**Comment by VizuaaLOG** [comment]
@nathanfranke :female_facepalm: was trying to fix this for so long I didn't think about the node's name property!

As you said there is still a bug here though as in this case the node name doesn't match a path. Not a complete waste of a few hours in that case :smile:

**Comment by nathanfranke** [comment]
Yes, there is a bug but it has nothing to do with deleting files. The problem is that there was no working Directory "open" check and the default directory is `res://`, so deleting files from what you think is `user://` actually deletes from `res://`.

PS: I really really hope this didn't destroy your project :frowning_face: Source control (git) has saved my life many times

**Comment by VizuaaLOG** [comment]
Ah I see, I guess in my case it just looked like delete because that's what I was doing but writing would have the same outcome.

I didnt lose anything luckily, also have stuff in git and although the code I wrote to delete the save deleted the code I wrote, it was nothing VSCode couldn't restore with a nice undo haha. Love VSCode for that! Anyway glad to see this is being included as a patch release for 3.2.

Thanks all for the speedy fix :laughing:
